packages/access-api/test/provider-add.test.js

```diff
@@ -0,0 +1,293 @@
+import {
+  assertNotError,
+  createTesterFromContext,
+  createTesterFromHandler,
+  warnOnErrorResult,
+} from './helpers/ucanto-test-utils.js'
+import * as principal from '@ucanto/principal'
+import * as provider from '@web3-storage/capabilities/provider'
+import * as assert from 'assert'
+import { createProviderAddHandler } from '../src/service/provider-add.js'
+import { context } from './helpers/context.js'
+import * as ucanto from '@ucanto/core'
+import * as Ucanto from '@ucanto/interface'
+import { Access, Provider } from '@web3-storage/capabilities'
+import * as delegationsResponse from '../src/utils/delegations-response.js'
+import { createStorageProvisions } from '../src/models/provisions.js'
+import { Email } from '../src/utils/email.js'
+
+for (const providerAddHandlerVariant of /** @type {const} */ ([
+  {
+    name: 'handled by createProviderAddHandler',
+    ...(() => {
+      const spaceWithStorageProvider = principal.ed25519.generate()
+      const storageProvisions = createStorageProvisions()
+      return {
+        spaceWithStorageProvider,
+        storageProvisions,
+        ...createTesterFromHandler(() =>
+          createProviderAddHandler({
+            storageProvisions,
+          })
+        ),
+      }
+    })(),
+  },
+])) {
+  describe(`provider/add ${providerAddHandlerVariant.name}`, () => {
+    it(`can be invoked by did:key`, async () => {
+      const space = await principal.ed25519.generate()
+      const issuer = await providerAddHandlerVariant.issuer
+      const result = await providerAddHandlerVariant.invoke(
+        await provider.add
+          .invoke({
+            issuer,
+            audience: await providerAddHandlerVariant.audience,
+            with: `did:mailto:example.com:foo`,
+            nb: {
+              consumer: space.did(),
+              provider: 'did:web:web3.storage:providers:w3up-alpha',
+            },
+          })
+          .delegate()
+      )
+      assertNotError(result)
+    })
+    it(`can be invoked by did:mailto w/ session ./update`, async () => {
+      const agentA = await principal.ed25519.generate()
+      const accountDID = /** @type {Ucanto.DID<'mailto'>} */ (
+        'did:mailto:example.com:foo'
+      )
+      const account = { did: () => accountDID }
+      const space = await principal.ed25519.generate()
+      const service = await providerAddHandlerVariant.audience
+      const agentACanSignAsAccountDID = await ucanto.delegate({
```

Comment:
I'm sure you know this, but this is no longer going to work. Instead you'd have to go through `access.Authorize` or alternatively create delegation from `absentee` account and then attestation from the `service` if you want to simplify the test.

Reply:
It seems that `testAuthorizeClaimProviderAdd` basically tests the same thing so I think you may want to remove this test entirely.

Reply:
> I'm sure you know this, but this is no longer going to work

Why won't it work?
It seems to work in the tests incl after #477 and with latest ucanto.

> It seems that testAuthorizeClaimProviderAdd basically tests the same thing so I think you may want to remove this test entirely.

I'm down to do that. It is slightly different because that other test doesn't explicitly test that `./update` works (even if that's currently what comes out of `access/claim`).

Reply:
I'm not sure why this is passing, but ucanto no longer knows anything about `./update`. In fact now that you've pointed out it is passing tests, I'm concerned that it does as I would expect it to fail instead.

Reply:
We looked at this together and realized it passed because this test only tests the 'handler' which is not wrapped in a ucanto `Server.provide` that would have checked the sigs.

I removed the test cuz its misleading. https://github.com/web3-storage/w3protocol/pull/462/commits/69038913304ec4e7d488bdde5846f24cedebfd88